### PR TITLE
Ignores reassigned and downstream if acknowledged

### PR DESCRIPTION
**Pull request by akhileshpillai**
Bug 1301129 - in perfherder alerts reassigned are marked as investigating
@wlach @jmaher

https://bugzilla.mozilla.org/show_bug.cgi?id=1301129
Please have a look.
I have made the changes to test and code.
Unfortunately I could'nt run the test
I tried the following from under the repo home
pytest
./manage.py test tests
./manage.py test --pattern=tests/perfalert/test_alert_modification.py

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1856)

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by wlach** [comment]
Hey Akhil, I will look at this in more detail soon, but meanwhile you can find instructions on running the tests here: http://treeherder.readthedocs.io/common_tasks.html#running-the-tests

**Comment by akhileshpillai** [comment]
I was trying to fix the test, it seems calling @pytest.fixture reference say for ex `test_per_alert`, always returns the same instance and I think I need multiple instances of `test_perf_alert` to include it in summary alert.
I found this online: 
http://stackoverflow.com/questions/21583833/multiple-copies-of-a-py-test-fixture
However, before doing anything else wanted to make sure I am not reinventing the wheel.  Please let me know if there is a way to get multiple instances of `test_perf_alert `

```
# ignore downstream and reassigned to update the summary status
    a.status = PerformanceAlert.REASSIGNED
    a.related_summary = s
    a.save()
    b = PerformanceAlert.objects.create(
            summary=s,
            series_signature=test_perf_signature,
            is_regression=True,
            amount_pct=0.5,
            amount_abs=50.0,
            prev_value=100.0,
            new_value=150.0,
            t_value=20.0)
    b.status = PerformanceAlert.ACKNOWLEDGED
    b.save()
    s = PerformanceAlertSummary.objects.get(id=1)
    assert s.status == PerformanceAlertSummary.IMPROVEMENT
```
the above fails with following error message:
```
if isinstance(errorvalue, BaseException):
>           raise errorvalue
E           IntegrityError: (1062, "Duplicate entry '1-1' for key 'performance_alert_summary_id_438059dd79175615_uniq'")
```

**Comment by wlach** [comment]
I wouldn't use a performance alert fixture for this test, just create the performance alert objects that you need inside the test itself. You're getting the error because you're violating the unique constraint on the performance summary, which disallows two alerts with the same signature to have the same summary.

File: treeherder/perf/models.py [hunk 1]
```diff
@@ -181,7 +181,8 @@ def autodetermine_status(self):
         # if not one of the resolved statuses and there are regressions,
         # otherwise we'll say it's an improvement
         if len([a for a in alerts if a.status == PerformanceAlert.ACKNOWLEDGED]):
-            if all([not alert.is_regression for alert in alerts]):
+            if all([not alert.is_regression for alert in
+                    alerts if alert.status == PerformanceAlert.ACKNOWLEDGED]):
```
Comment (wlach) [inline treeherder/perf/models.py]: This looks right!

File: tests/perfalert/test_alert_modification.py [hunk 2]
```diff
@@ -27,6 +27,18 @@ def test_summary_modification(test_repository, test_perf_signature,
...
+    s = PerformanceAlertSummary.objects.get(id=1)
+    s.status = PerformanceAlertSummary.INVESTIGATING
+    s.save()
+    a.status = PerformanceAlert.REASSIGNED
+    a.save()
+    b = test_perf_alert
+    b.status = PerformanceAlert.ACKNOWLEDGED
+    b.save()
+    s = PerformanceAlertSummary.objects.get(id=1)
+    assert s.status == PerformanceAlertSummary.IMPROVEMENT
```
Comment (wlach) [inline tests/perfalert/test_alert_modification.py]: It's probably worth breaking this out into a seperate test (see the comments in the discussion)

Comment (akhileshpillai) [inline tests/perfalert/test_alert_modification.py]: Sounds Good.

**Comment by akhileshpillai** [comment]
Just saw the review comments, Unforutnately the email setting was pointing to my change.org address. Fixed it.

File: tests/perfalert/test_alert_modification.py [hunk 3]
```diff
@@ -1,10 +1,35 @@
...
 import pytest
-from django.core.exceptions import ValidationError
 
+from django.core.exceptions import ValidationError
 from treeherder.perf.models import (PerformanceAlert,
                                     PerformanceAlertSummary)
+from treeherder.model.models import (MachinePlatform,
+                                     Option,
+                                     OptionCollection)
+from treeherder.perf.models import PerformanceSignature
```
Comment (wlach) [inline tests/perfalert/test_alert_modification.py]: The ordering here is generating a travis linter error: https://travis-ci.org/mozilla/treeherder/jobs/161086026

You can find out how to run the linters yourself here: http://treeherder.readthedocs.io/common_tasks.html#running-the-tests

File: tests/perfalert/test_alert_modification.py [hunk 4]
```diff
@@ -28,6 +53,76 @@ def test_summary_modification(test_repository, test_perf_signature,
...
+        repository=repo_ref('default', "dev2", "test_treeherder_2"),
+        signature_hash=(40*'t'),
+        framework=test_perf_framework,
+        platform=platform_2,
+        option_collection=option_collection_2,
+        suite='mysuite_2',
+        test='mytest_2',
+        has_subtests=False,
+        last_updated=datetime.datetime.now()
+    )
```
Comment (wlach) [inline tests/perfalert/test_alert_modification.py]: I think you should be able to get away without most of this. You should be able to just use `test_repository` for the repository for all signatures. For the first signature, just use `test_perf_signature`. For the second, you can inherit most of its properties, just rename the suite and test. E.g.:


```py
signature2 = PerformanceSignature.objects.create(
  repository=test_repository,
  signature_hash=(40*'u'),
  framework=test_perf_signature.framework,
  platform = test_perf_signature.platfomr,
  option_collection=test_perf_signature.platform
  suite='mysuite_2',
  test='mytest_2',
  has_subtests=False,
  last_updated=datetime.datetime.now())
```

File: tests/perfalert/test_alert_modification.py [hunk 5]
```diff
@@ -28,6 +53,76 @@ def test_summary_modification(test_repository, test_perf_signature,
...
+            is_regression=True,
+            amount_pct=0.5,
+            amount_abs=50.0,
+            prev_value=100.0,
+            new_value=150.0,
+            t_value=20.0)
+    a.status = PerformanceAlert.REASSIGNED
+    a.related_summary = s
+    a.save()
+    b = PerformanceAlert.objects.create(
```
Comment (wlach) [inline tests/perfalert/test_alert_modification.py]: Add a comment above this line to say this is the test case.

File: tests/perfalert/test_alert_modification.py (hunk 4 above)
Comment (akhileshpillai) [inline tests/perfalert/test_alert_modification.py]: I will try this now

**Pull request merged**